subprojects/build-events/src/main/java/org/gradle/internal/build/event/types/DefaultResourceBasedTestDescriptor.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.build.event.types;
+
+import org.gradle.internal.operations.OperationIdentifier;
+import org.gradle.tooling.internal.protocol.events.InternalOperationDescriptor;
+import org.gradle.tooling.internal.protocol.events.InternalResourceBasedTestDescriptor;
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
+
+import java.io.File;
+import java.io.Serializable;
+
+@NullMarked
+public class DefaultResourceBasedTestDescriptor extends DefaultTestDescriptor implements Serializable, InternalResourceBasedTestDescriptor, InternalOperationDescriptor {
```

Comment:
❓ Do we have any option for not extending `DefaultTestDescriptor` here?  This really shouldn't need a `suite/class/methodName` - which I see are nullable, but still, the inheritance itself is misleading.

I see on `InternalJvmTestDescriptor` the note "DO NOT CHANGE THIS INTERFACE. It is part of the cross-version protocol."